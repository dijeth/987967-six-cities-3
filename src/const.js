const OfferType = {
  APARTMENT: `Apartment`,
  PRIVATE_ROOM: `Private room`
};

const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export {OfferType, CITIES};
