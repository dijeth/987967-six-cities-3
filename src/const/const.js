"use strict";
exports.__esModule = true;
exports.OfferType = {
    APARTMENT: "Apartment",
    PRIVATE_ROOM: "Private room"
};
exports.CITIES = [
    "Paris",
    "Cologne",
    "Brussels",
    "Amsterdam",
    "Hamburg",
    "Dusseldorf"
];
exports.InsideFeature = {
    WI_FI: "Wi-Fi",
    WASHING_MACHINE: "Washing machine",
    TOWELS: "Towels",
    HEATING: "Heating",
    COFFEE_MACHINE: "Coffee machine",
    BABY_SEAT: "Baby seat",
    KITCHEN: "Kitchen",
    DISHWASHER: "Dishwasher",
    CABEL_TV: "Cabel TV",
    FRIDGE: "Fridge"
};
exports.ScreenType = {
    MAIN: "MAIN",
    PROPERTY: "PROPERTY"
};
exports.BREAK_STRING = "\n";
exports.MONTHS = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
exports.MAX_CITY_COUNT = 6;
var SortType;
(function (SortType) {
    SortType["POPULAR"] = "Popular";
    SortType["PRICE_LOW_TO_HIGH"] = "Price: low to high";
    SortType["PRICE_HIGH_TO_LOW"] = "Price: high to low";
    SortType["TOP_RATED_FIRST"] = "Top rated first";
})(SortType = exports.SortType || (exports.SortType = {}));
exports.SORT_LIST = [
    SortType.POPULAR,
    SortType.PRICE_LOW_TO_HIGH,
    SortType.PRICE_HIGH_TO_LOW,
    SortType.TOP_RATED_FIRST
];
exports.AuthorizationStatus = {
    NO_AUTH: "NO_AUTH",
    AUTH: "AUTH"
};
exports.AppRoute = {
    getLogin: function () { return "/login"; },
    getOffer: function (id) { return "/offer/" + id; },
    getFavorites: function () { return "/favorites"; },
    getRoot: function () { return "/"; }
};
exports.ServerRoute = {
    getLogin: function () { return "/login"; },
    getFavorites: function () { return "/favorite"; },
    postFavorites: function (id, status) { return "/favorite/" + id + "/" + status; },
    getNearby: function (id) { return "/hotels/" + id + "/nearby"; },
    getHotels: function () { return "/hotels"; },
    getComments: function (id) { return "/comments/" + id; }
};
exports.MAX_IMAGE_COUNT = 6;
exports.ReviewLength = {
    MIN: 50,
    MAX: 300
};
exports.ServerError = {
    UNAUTHORIZED: 401,
    BAD_REQUEST: 400,
    NOT_FOUND: 404
};
var PlaceCardType;
(function (PlaceCardType) {
    PlaceCardType["DEFAULT"] = "DEFAULT";
    PlaceCardType["FAVORITE"] = "FAVORITE";
    PlaceCardType["NEARBY"] = "NEARBY";
})(PlaceCardType = exports.PlaceCardType || (exports.PlaceCardType = {}));
exports.EMPTY_REVIEW = {
    text: "",
    rating: 0,
    offerID: null
};
exports.BASE_URL = "https://htmlacademy-react-3.appspot.com/six-cities";
exports.TIME_OUT = 5000;
