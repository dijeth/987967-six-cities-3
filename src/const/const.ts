export const OfferType = {
  APARTMENT: `Apartment`,
  PRIVATE_ROOM: `Private room`
};

export const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const InsideFeature = {
  WI_FI: `Wi-Fi`,
  WASHING_MACHINE: `Washing machine`,
  TOWELS: `Towels`,
  HEATING: `Heating`,
  COFFEE_MACHINE: `Coffee machine`,
  BABY_SEAT: `Baby seat`,
  KITCHEN: `Kitchen`,
  DISHWASHER: `Dishwasher`,
  CABEL_TV: `Cabel TV`,
  FRIDGE: `Fridge`
};

export const ScreenType = {
  MAIN: `MAIN`,
  PROPERTY: `PROPERTY`
};

export const BREAK_STRING = `\n`;

export const MONTHS = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];

export const MAX_CITY_COUNT = 6;

export enum SortType {
  POPULAR = `Popular`,
  PRICE_LOW_TO_HIGH = `Price: low to high`,
  PRICE_HIGH_TO_LOW = `Price: high to low`,
  TOP_RATED_FIRST = `Top rated first`,
};

export const SORT_LIST = [
  SortType.POPULAR,
  SortType.PRICE_LOW_TO_HIGH,
  SortType.PRICE_HIGH_TO_LOW,
  SortType.TOP_RATED_FIRST
];

export const AuthorizationStatus = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`,
};

export const AppRoute = {
  getLogin: () => `/login`,
  getOffer: (id) => `/offer/${id}`,
  getFavorites: () => `/favorites`,
  getRoot: () => `/`,
};

export const ServerRoute = {
  getLogin: () => `/login`,
  getFavorites: () => `/favorite`,
  postFavorites: (id, status) => `/favorite/${id}/${status}`,
  getNearby: (id) => `/hotels/${id}/nearby`,
  getHotels: () => `/hotels`,
  getComments: (id) => `/comments/${id}`,
};

export const MAX_IMAGE_COUNT = 6;

export const ReviewLength = {
  MIN: 50,
  MAX: 300
};

export const ServerError = {
  UNAUTHORIZED: 401,
  BAD_REQUEST: 400,
  NOT_FOUND: 404
};

export enum PlaceCardType {
  DEFAULT = `DEFAULT`,
  FAVORITE = `FAVORITE`,
  NEARBY = `NEARBY`,
};

export const EMPTY_REVIEW = {
  text: ``,
  rating: 0,
  offerID: null,
};

export const BASE_URL = `https://htmlacademy-react-3.appspot.com/six-cities`;
export const TIME_OUT = 5000;
