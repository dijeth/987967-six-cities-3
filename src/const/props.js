import PropTypes from 'prop-types';

export const offerPropType = PropTypes.shape({
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  pictures: PropTypes.arrayOf(PropTypes.string).isRequired,
  cost: PropTypes.number.isRequired,
  rating: PropTypes.number.isRequired,
  isPremium: PropTypes.bool.isRequired,
  isFavorite: PropTypes.bool.isRequired,
  city: PropTypes.string.isRequired,
  coord: PropTypes.arrayOf(PropTypes.number).isRequired
});

export const cityPropType = PropTypes.shape({
  name: PropTypes.string.isRequired,
  zoom: PropTypes.number.isRequired,
  centerCoord: PropTypes.arrayOf(PropTypes.number)
});
