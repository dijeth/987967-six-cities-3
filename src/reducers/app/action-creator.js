import ActionType from '../action-type.js';

const ActionCreator = {
  changeCity: (city) => ({type: ActionType.CHANGE_CITY, payload: city}),
  changeActiveOffer: (offer) => ({type: ActionType.CHANGE_ACTIVE_OFFER, payload: offer}),
  changeScreenType: (screenType) => ({type: ActionType.CHANGE_SCREEN_TYPE, payload: screenType}),
  changeSortType: (sortType) => ({type: ActionType.CHANGE_SORT_TYPE, payload: sortType}),
};

export default ActionCreator;
