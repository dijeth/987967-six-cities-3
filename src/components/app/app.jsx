import React from 'react';
import PropTypes from 'prop-types';
import {BrowserRouter, Route, Switch} from 'react-router-dom';
import Main from '../main/main.jsx';
import CardProperty from '../card-property/card-property.jsx';
import {getNeighbourhoods} from '../../mocks/offers.js';
import {connect} from 'react-redux';
import {ScreenType} from '../../const.js';
import {offerPropType} from '../place-card/place-card.jsx';
import withSort from '../../hocs/with-sort/with-sort.jsx';

const MainWithSort = withSort(Main);

const App = ({screenType, activeOffer, offers}) => {
  const isNearPlaces = screenType === ScreenType.PROPERTY;

  if (screenType === ScreenType.PROPERTY) {
    const neighbourhoods = getNeighbourhoods(activeOffer, offers);
    return <CardProperty
      offer={activeOffer}
      neighbourhoods={neighbourhoods}
      isNearPlaces={isNearPlaces}
    />;
  }

  return (
    <BrowserRouter>
      <Switch>
        <Route exact path="/">
          <MainWithSort
            isNearPlaces={screenType === ScreenType.PROPERTY}
          />
        </Route>
        <Route exact path="/dev-card-property">
          <CardProperty
            offer={offers[0]}
            neighbourhoods={getNeighbourhoods(offers[0], offers)}
            isNearPlaces={isNearPlaces}
          />
        </Route>
      </Switch>
    </BrowserRouter>);
};

App.propTypes = {
  screenType: PropTypes.oneOf([ScreenType.MAIN, ScreenType.PROPERTY]).isRequired,
  activeOffer: offerPropType,
  offers: PropTypes.arrayOf(offerPropType)
};

const mapStateToProps = (state) => ({
  screenType: state.screenType,
  activeOffer: state.activeOffer,
  offers: state.offers
});

export {App};
export default connect(mapStateToProps)(App);
