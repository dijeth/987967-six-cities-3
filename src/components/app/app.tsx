import * as React from 'react';
import {Router, Route, Switch} from 'react-router-dom';
import PageMain from '../page-main/page-main';
import PageFavotites from '../page-favorites/page-favorites';
import PageProperties from '../page-properties/page-properties';
import PageSignIn from '../page-sign-in/page-sign-in';
import history from '../../history.js';
import withPathName from '../../hocs/with-pathname/with-pathname.jsx';
import {getAuthorizationStatus} from '../../reducers/user/selectors.js';
import {connect} from 'react-redux';
import PrivateRoute from '../private-route/private-route.jsx';
import {AppRoute} from '../../const/const.ts';

const PagePropertiesWithPathName = withPathName(PageProperties);

const App: React.FC<{isAuth: boolean}> = ({isAuth}) => {
  return (
    <Router history={history}>
      <Switch>
        <Route exact path={AppRoute.getOffer(`:id`)} render={({match}) => {
          const id = match.params.id;
          return <PagePropertiesWithPathName pathID={id} />;
        }} />
        <PrivateRoute path={AppRoute.getLogin()} require={!isAuth} to={AppRoute.getRoot()} render={
          () => <PageSignIn />
        } />
        <PrivateRoute path={AppRoute.getFavorites()} require={isAuth} to={AppRoute.getLogin()} render={
          () => <PageFavotites />
        } />
        <Route exact path={AppRoute.getRoot()} component={PageMain} />
        <PageMain />
      </Switch>
    </Router>);
};

const mapStateToProps = (state) => ({
  isAuth: getAuthorizationStatus(state)
});

export {App};
export default connect(mapStateToProps)(App);
