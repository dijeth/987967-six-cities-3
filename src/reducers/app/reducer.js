import {SortType} from '../../const/const.js';
import ActionType from '../action-type.js';

const initialState = {
  activeCity: null,
  activeOffer: null,
  sortType: SortType.POPULAR,
  loading: 0,
  isCommentSending: false,
  commentError: false,
  pageError: ``
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_CITY:
      return Object.assign({}, state, {activeCity: action.payload});

    case ActionType.CHANGE_ACTIVE_OFFER:
      return Object.assign({}, state, {activeOffer: action.payload});

    case ActionType.CHANGE_SORT_TYPE:
      return Object.assign({}, state, {sortType: action.payload});

    case ActionType.INCREASE_LOAD:
      return Object.assign({}, state, {loading: state.loading+1});

    case ActionType.DECREASE_LOAD:
      return Object.assign({}, state, {loading: state.loading-1});

    case ActionType.CHANGE_COMMENT_SENDING_STATUS:
      return Object.assign({}, state, {isCommentSending: action.payload});

    case ActionType.SET_COMMENT_ERROR:
      return Object.assign({}, state, {commentError: action.payload});

    case ActionType.SET_PAGE_ERROR:
      return Object.assign({}, state, {pageError: action.payload});
  }

  return state;
};

export default reducer;
