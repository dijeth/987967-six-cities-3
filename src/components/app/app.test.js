import React from 'react';
import renderer from 'react-test-renderer';
import App from './app.jsx';

const PLACES_COUNT = 100;
const PLACE_CARD_NAMES = [`placeCardName1`, `placeCardName2`];

it(`<App /> should be render correctly`, () => {
  const app = renderer.create(<App placesCount={PLACES_COUNT} placeCardNames={PLACE_CARD_NAMES}/>).toJSON();
  expect(app).toMatchSnapshot();
});
