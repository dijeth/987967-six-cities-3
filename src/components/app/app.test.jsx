import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import {App} from './app.jsx';
import NameSpace from '../../reducers/name-space.js';
import {SortType} from '../../const/const.js';

const mockStore = configureStore([]);

const store = mockStore({
  [NameSpace.USER]: {
    authorizationStatus: `NO_AUTH`,
    authInfo: null,
  },

  [NameSpace.DATA]: {
    cities: [{
      name: `Dusseldorf`,
      zoom: 13,
      centerCoord: [51.225402, 6.776314]
    }, {
      name: `Brussels`,
      zoom: 13,
      centerCoord: [50.846557, 4.351697]
    }],

    offers: [{
      id: `1`,
      city: `Dusseldorf`,
      pictures: [`https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg`,
        `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg`,
        `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/5.jpg`,
      ],
      title: `Nice, cozy, warm big bed apartment`,
      descriptionTitle: `Meet the host`,
      isFavorite: false,
      isPremium: true,
      rating: 3.5,
      type: `house`,
      bedroomCount: 2,
      adultsCount: 4,
      cost: 868,
      insideFeatures: [
        `Laptop friendly workspace`,
        `Breakfast`,
        `Baby seat`,
        `Towels`,
        `Washer`,
        `Air conditioning`,
      ],
      userName: `Angelina`,
      userPicture: `img/avatar-angelina.jpg`,
      isSuperUser: true,
      userID: `25`,
      description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      coord: [51.237402,
        6.797314
      ],
      zoom: 16
    }, {
      id: `2`,
      city: `Brussels`,
      pictures: [`https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/13.jpg`,
        `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/1.jpg`,
        `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/16.jpg`,
        `https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/3.jpg`
      ],
      title: `The Pondhouse - A Magical Place`,
      descriptionTitle: `Meet the host`,
      isFavorite: false,
      isPremium: false,
      rating: 4.5,
      type: `room`,
      bedroomCount: 1,
      adultsCount: 3,
      cost: 270,
      insideFeatures: [`Washer`,
        `Laptop friendly workspace`,
        `Air conditioning`,
        `Breakfast`
      ],
      userName: `Angelina`,
      userPicture: `img/avatar-angelina.jpg`,
      isSuperUser: true,
      userID: `25`,
      description: `Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.`,
      coord: [50.869557,
        4.332697
      ],
      zoom: 16
    }]
  },

  [NameSpace.APP]: {
    pageError: ``,
    loading: 0,
    activeCity: `Dusseldorf`,
    activeOffer: null,
    sortType: SortType.POPULAR,
  }
});

it(`1`, () => {
  const tree = renderer.create(
      <Provider store={store}>
        <App isAuth={true} />
      </Provider>,
      {createNodeMock: () => document.createElement(`div`)}
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
