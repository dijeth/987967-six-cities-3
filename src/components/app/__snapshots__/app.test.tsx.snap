// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 1`] = `
<div
  class="page page--gray page--main"
>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link header__logo-link--active"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
          >
            <li
              class="header__nav-item user"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/login"
              >
                <div
                  class="header__avatar-wrapper  user__avatar-wrapper"
                />
                <span
                  class="header__login"
                >
                  Sign in
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    class="page__main page__main--index "
  >
    <h1
      class="visually-hidden"
    >
      Cities
    </h1>
    <div
      class="tabs"
    >
      <section
        class="locations container"
      >
        <ul
          class="locations__list tabs__list"
        >
          <li
            class="locations__item"
          >
            <a
              class="locations__item-link tabs__item tabs__item--active"
              href="#"
            >
              <span
                data-index="0"
              >
                Dusseldorf
              </span>
            </a>
          </li>
          <li
            class="locations__item"
          >
            <a
              class="locations__item-link tabs__item "
              href="#"
            >
              <span
                data-index="1"
              >
                Brussels
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      class="cities"
    >
      <div
        class="cities__places-container container"
      >
        <section
          class="cities__places places"
        >
          <h2
            class="visually-hidden"
          >
            Places
          </h2>
          <b
            class="places__found"
          >
            1
             places to stay in 
            Dusseldorf
          </b>
          <form
            action="#"
            class="places__sorting"
            method="get"
          >
            <span
              class="places__sorting-caption"
            >
              Sort by
            </span>
            <span
              class="places__sorting-type"
              tabindex="0"
            >
              Popular
              <svg
                class="places__sorting-arrow"
                height="4"
                width="7"
              >
                <use
                  xlink:href="#icon-arrow-select"
                />
              </svg>
            </span>
            <ul
              class="places__options places__options--custom "
            >
              <li
                class="places__option places__option--active"
                tabindex="0"
              >
                Popular
              </li>
              <li
                class="places__option "
                tabindex="0"
              >
                Price: low to high
              </li>
              <li
                class="places__option "
                tabindex="0"
              >
                Price: high to low
              </li>
              <li
                class="places__option "
                tabindex="0"
              >
                Top rated first
              </li>
            </ul>
          </form>
          <div
            class="cities__places-list tabs__content places__list"
          >
            <article
              class="cities__place-card place-card"
            >
              <div
                class="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                class="cities__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="#"
                >
                  <img
                    alt="Place image"
                    class="place-card__image"
                    height="200"
                    src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/15.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                class=" place-card__info"
              >
                <div
                  class="place-card__price-wrapper"
                >
                  <div
                    class="place-card__price"
                  >
                    <b
                      class="place-card__price-value"
                    >
                      €
                      868
                    </b>
                    <span
                      class="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <a
                    class="place-card__bookmark-button button"
                    href="/login"
                  >
                    <svg
                      class="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlink:href="#icon-bookmark"
                      />
                    </svg>
                    <span
                      class="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </a>
                </div>
                <div
                  class="place-card__rating rating"
                >
                  <div
                    class="place-card__stars rating__stars"
                  >
                    <span
                      style="width: 80%;"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  class="place-card__name"
                >
                  <a
                    href="/offer/1"
                  >
                    Nice, cozy, warm big bed apartment
                  </a>
                </h2>
                <p
                  class="place-card__type"
                >
                  house
                </p>
              </div>
            </article>
          </div>
        </section>
        <div
          class="cities__right-section"
        >
          <section
            class="cities__map map"
          >
            <div
              class="leaflet-container leaflet-grab leaflet-touch-drag"
              id="map"
              style="width: 100%; height: 100%; position: relative;"
              tabindex="0"
            >
              <div
                class="leaflet-pane leaflet-map-pane"
                style="left: 0px; top: 0px;"
              >
                <div
                  class="leaflet-pane leaflet-tile-pane"
                >
                  <div
                    class="leaflet-layer "
                    style="z-index: 1;"
                  >
                    <div
                      class="leaflet-tile-container leaflet-zoom-animated"
                      style="z-index: 18; left: 0px; top: 0px;"
                    >
                      <img
                        alt=""
                        class="leaflet-tile"
                        role="presentation"
                        src="https://a.basemaps.cartocdn.com/rastertiles/voyager_labels_under/13/4250/2734{@2x}.png"
                        style="width: 256px; height: 256px; left: -51px; top: -84px;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="leaflet-pane leaflet-shadow-pane"
                />
                <div
                  class="leaflet-pane leaflet-overlay-pane"
                />
                <div
                  class="leaflet-pane leaflet-marker-pane"
                >
                  <img
                    alt=""
                    class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                    src="img/pin.svg"
                    style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: 122px; top: -112px; z-index: -112;"
                    tabindex="0"
                  />
                </div>
                <div
                  class="leaflet-pane leaflet-tooltip-pane"
                />
                <div
                  class="leaflet-pane leaflet-popup-pane"
                />
              </div>
              <div
                class="leaflet-control-container"
              >
                <div
                  class="leaflet-top leaflet-left"
                />
                <div
                  class="leaflet-top leaflet-right"
                />
                <div
                  class="leaflet-bottom leaflet-left"
                />
                <div
                  class="leaflet-bottom leaflet-right"
                >
                  <div
                    class="leaflet-control-attribution leaflet-control"
                  >
                    <a
                      href="https://leafletjs.com"
                      title="A JS library for interactive maps"
                    >
                      Leaflet
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
    </div>
  </main>
</div>
`;
